src/main/java/teammates/sqllogic/core/FeedbackSessionsLogic.java

```diff
@@ -386,4 +389,24 @@ public void adjustFeedbackSessionEmailStatusAfterUpdate(FeedbackSession session)
             session.setPublishedEmailSent(session.isPublished());
         }
     }
+
+    /**
+     * Returns a list of sessions that were closed within past hour.
+     */
+    public List<FeedbackSession> getFeedbackSessionsClosedWithinThePastHour() {
+        List<FeedbackSession> requiredSessions = new ArrayList<>();
+        List<FeedbackSession> sessions = fsDb.getFeedbackSessionsPossiblyNeedingClosedEmail();
+        log.info(String.format("Number of sessions under consideration: %d", sessions.size()));
+
+        for (FeedbackSession session : sessions) {
+            // is session closed in the past 1 hour
+            if (session.isClosedWithinPastHour()
```

Comment:
Is it not possible to do this check through Hibernate?

Reply:
We could, but I think this restructures some of the existing logic and might change the semantics for what a method should be doing. Maybe leave this for a future optimization?

Reply:
> We could, but I think this restructures some of the existing logic and might change the semantics for what a method should be doing. Maybe leave this for a future optimization?

+1 since we're quite pressed for time, the original way of implementation will do for now - let's focus more on correctness